subsys/storage/stream/stream_flash.c

```diff
@@ -42,6 +41,8 @@ int stream_flash_erase_page(struct stream_flash_ctx *ctx, off_t off)
 
 	if (rc != 0) {
 		LOG_ERR("Error %d while erasing page", rc);
+	} else {
+		ctx->last_erased_page_start_offset = page.start_offset;
```

Comment:
Could we add a unit test case for this?

Reply:
Yes, we should. Sorry I have completely forgotten to write tests for this while stuck on on other tasks. I will add tests for the changes.